Chores/CS and linter fixes backend

devmount opened the pull request:
<!--
* Filling out the template is required.
* All new code must have been tested to ensure against regressions
-->

## Description of the Change

This applies `black` and `ruff` on our current backend. Seems like we didn't do that for quite some time 😅 Let me know, if you want to change any rules here, afaik we're just applying default rules and a few in pyproject.toml

## Benefits

Consistent code base

## Applicable Issues

None

devmount commented on backend/pyproject.toml:
```diff
@@ -73,7 +73,7 @@ line-length = 120
 dummy-variable-rgx = "^(_+|(_+[a-zA-Z0-9_]*[a-zA-Z0-9]+?))$"
 
-# Assume Python 3.10.
+# TODO: Assume Python 3.10.
```
@MelissaAutumn Do we need to update this? I did this PR on Python 3.12 with current versions of black and ruff...

MelissaAutumn replied:
We can at least upgrade this to 3.11. I think 3.12 is fine, but 3.11 is the docker container's target.

devmount replied:
Alright, changed that. Thanks!

MelissaAutumn commented:
Thanks for the config fixes! Seems like we might not need black anymore since ruff's formatter sells itself as a drop-in replacement. 

We should switch the quote style back to single quotes and try and fix the excessive param exploding when it comes to functions that contain a dictionary or list as its sole argument.

devmount commented:
I removed black and added a ruff rule to use single quotes (but somehow the don't get applied on fix?). Do you know which rule to add to fix the function parameters? Thanks!

devmount commented:
Ok for some reason the quotes config in pyproject.toml wasn't taken into account, so I manually ran this now:

```bash
ruff format backend --config "format.quote-style = 'single'"
```

devmount commented:
I actually tried that too without success, but I can add it to the pyproject.toml nevertheless

devmount commented:
Ok, I now added the inline quotes rule as well.

MelissaAutumn commented:
Thanks!

devmount commented on backend/src/appointment/dependencies/auth.py:
```diff
@@ -18,24 +18,28 @@
-        subscriber and subscriber.minimum_valid_iat_time and subscriber.minimum_valid_iat_time.timestamp() > int(iat)
-    ]):
+    if any(
+        [
```
I agree, but I couldn't find a rule for that...
